core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -486,12 +487,14 @@ private[kafka] class Processor(val id: Int,
   private def processCompletedReceives() {
     selector.completedReceives.asScala.foreach { receive =>
       try {
-        val channel = selector.channel(receive.source)
+        val openChannel = selector.channel(receive.source)
+        val channel = if (openChannel != null) openChannel else selector.closedChannel(receive.source)
```

Comment:
@ijuma Thank you, I have updated the code. I had seen that JIRA, but couldn't figure out the scenario either.